Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -2,5 +2,59 @@
 
 void trainingTask(uint8_t* data)
 {
+	validData = 0;
+	
+	if (motorCheck(*data) && motorCheck(*data) == motorCheck(*(data + 1)) && lightsCheck(*(data + 2)) ) // set outputArray if motors data are valid, motors are in sync and lights data is valid
+	{
+		for (int i = 0; i < sizeof outputArray / sizeof *outputArray; i++)
```

Comment:
Do `sizeof(outputArray)/sizeof(*outputArray)`. You could also just hardcode it to 3